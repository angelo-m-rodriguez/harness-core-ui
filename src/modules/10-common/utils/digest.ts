/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

async function digest(str: string): Promise<string> {
  const msgUint8 = new TextEncoder().encode(str) // encode as (utf-8) Uint8Array
  const hashBuffer = await crypto.subtle.digest('SHA-256', msgUint8) // hash the message
  const hashArray = Array.from(new Uint8Array(hashBuffer)) // convert buffer to byte array
  const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('') // convert bytes to hex string

  return hashHex
}

export default digest
