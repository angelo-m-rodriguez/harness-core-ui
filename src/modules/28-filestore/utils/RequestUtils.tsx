/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FilterDataInterface, FilterInterface } from '@common/components/Filter/Constants'
import type { EmbeddedUserDetailsDTO, FilesFilterProperties, FilterDTO } from 'services/cd-ng'
import { StringUtils } from '@common/exports'

export type FileStoreFilterFormType = Omit<FilesFilterProperties, 'createdBy' | 'referencedBy'> & {
  createdBy?: string
  referencedBy?: string
}

export const createRequestBodyPayload = ({
  isUpdate,
  data,
  projectIdentifier,
  orgIdentifier,
  accountIdentifier,
  referenceIdentifier,
  createdByList
}: {
  isUpdate: boolean
  data: FilterDataInterface<FileStoreFilterFormType, FilterInterface>
  projectIdentifier: string
  orgIdentifier: string
  accountIdentifier: string
  referenceIdentifier?: string
  createdByList: EmbeddedUserDetailsDTO[]
}): FilterDTO => {
  const {
    metadata: { name: _name, filterVisibility, identifier },
    formValues
  } = data

  const createdBy = createdByList.find(user => formValues.createdBy === user.email)

  return {
    name: _name,
    identifier: isUpdate ? identifier : StringUtils.getIdentifierFromName(_name),
    projectIdentifier,
    orgIdentifier,
    filterVisibility: filterVisibility,
    filterProperties: {
      createdBy: createdBy,
      ...(formValues.referencedBy && referenceIdentifier
        ? {
            referencedBy: {
              type: formValues.referencedBy,
              entityRef: {
                identifier: referenceIdentifier,
                orgIdentifier,
                projectIdentifier,
                accountIdentifier
              }
            }
          }
        : {}),
      fileUsage: formValues.fileUsage,
      tags: formValues.tags,
      filterType: 'FileStore'
    } as FilesFilterProperties
  }
}
