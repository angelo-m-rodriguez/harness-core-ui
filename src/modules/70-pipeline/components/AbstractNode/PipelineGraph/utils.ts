import type { RefObject } from 'react'
import type { StageElementWrapperConfig } from 'services/cd-ng'
const INITIAL_ZOOM_LEVEL = 1
const ZOOM_INC_DEC_LEVEL = 0.1
const getFinalSVGArrowPath = (parentId: string, id1 = '', id2 = '', isParallelNode = false): string => {
  if (!parentId) {
    return ''
  }
  const node1 = getComputedPosition(parentId, id1)
  const node2 = getComputedPosition(parentId, id2)
  if (!node1 || !node2) {
    return ''
  }

  const node1VerticalMid = node1.top + node1.height / 2
  const node2VerticalMid = node2.top + node2.height / 2

  const startPoint = `${node1.right},${node1VerticalMid}`
  const horizontalMid = Math.abs((node1.right + node2.left) / 2)
  const endPoint = `${node2.left},${node2VerticalMid}`

  if (node2.y < node1.y) {
    //  child node is at top
    const curveLeftToTop = `Q${horizontalMid},${node1VerticalMid} ${horizontalMid},${node1VerticalMid - 20}`
    const curveBottomToRight = `Q${horizontalMid},${node2VerticalMid} ${horizontalMid + 20},${node2VerticalMid}`
    return `M${startPoint} L${horizontalMid - 20},${node1VerticalMid} ${curveLeftToTop} 
    L${horizontalMid},${node2VerticalMid + 20} ${curveBottomToRight} L${endPoint}`
  } else if (node1.y === node2.y) {
    // both nodes are at same level vertically
    return `M${startPoint}  L${endPoint} `
  } else {
    //  child node is at bottom
    const curveLeftToBottom = `Q${horizontalMid},${node1VerticalMid} ${horizontalMid},${node1VerticalMid + 20}`

    const curveTopToRight = `Q${horizontalMid},${node2VerticalMid} ${horizontalMid + 20},${node2VerticalMid}`

    if (isParallelNode) {
      const updatedStart = node1.left - 45 // new start point
      const parallelLinkStart = `${
        updatedStart // half of link length
      },${node1VerticalMid}`

      const curveLBparallel = `Q${updatedStart + 20},${node1VerticalMid} ${updatedStart + 20},${node1VerticalMid + 20} `
      const curveTRparallel = `Q${updatedStart + 20},${node2VerticalMid} ${updatedStart + 40},${node2VerticalMid}`

      return `M${parallelLinkStart} 
      ${curveLBparallel} 
      L${updatedStart + 20},${node2VerticalMid - 20} 
      ${curveTRparallel} 
      L${node2.left},${node2VerticalMid} 
      M${node2.right},${node2VerticalMid} 
      L${node2.right + 10},${node2VerticalMid}
      Q${node2.right + 25},${node2VerticalMid} ${node2.right + 25},${node2VerticalMid - 20}      
      L${node2.right + 25},${node1VerticalMid + 20}
      Q${node2.right + 25},${node1VerticalMid} ${node2.right + 40},${node1VerticalMid}
      `
    }
    return `M${startPoint} L${horizontalMid - 20},${node1VerticalMid} ${curveLeftToBottom} 
    L${horizontalMid},${node2VerticalMid - 20} ${curveTopToRight} L${endPoint}`
  }
}

const getComputedPosition = (parentId: string, childId: string): DOMRect => {
  const parentPos = document.getElementById(parentId)?.getBoundingClientRect() as DOMRect
  const childPos = document.getElementById(childId)?.getBoundingClientRect() as DOMRect
  const updatedTop = childPos.top - parentPos.top
  const updatedLeft = childPos.left - parentPos.left
  const updatedRight = updatedLeft + childPos.width
  const updatedBottom = updatedTop + childPos.height
  const updatedPos: DOMRect = {
    ...childPos,
    left: updatedLeft,
    top: updatedTop,
    right: updatedRight,
    bottom: updatedBottom,
    width: childPos.width,
    height: childPos.height,
    x: childPos.x,
    y: childPos.y
  }
  return updatedPos
}

const setupDragEventListeners = (canvasRef: RefObject<HTMLDivElement>): (() => void) => {
  let offset = [0, 0]
  const divOverlay = (canvasRef.current as HTMLDivElement).parentElement as HTMLDivElement
  const canvas = canvasRef.current as HTMLDivElement
  let isDown = false

  const onMouseMove = (e: any): void => {
    e.preventDefault()
    if (isDown) {
      canvas.style.left = e.clientX + offset[0] + 'px'
      canvas.style.top = e.clientY + offset[1] + 'px'
    }
  }

  const onMouseDown = (e: any): void => {
    isDown = true
    offset = [canvas.offsetLeft - e.clientX, canvas.offsetTop - e.clientY]
  }
  const onMouseUp = (): void => {
    isDown = false
  }

  divOverlay.addEventListener('mousedown', onMouseDown, true)
  document.addEventListener('mouseup', onMouseUp, true)
  document.addEventListener('mousemove', onMouseMove, true)
  return () => {
    divOverlay.removeEventListener('mousedown', onMouseDown)
    document.removeEventListener('mouseup', onMouseUp)
    document.removeEventListener('mousemove', onMouseMove)
  }
}

const getSVGLinksFromPipeline = (stages?: StageElementWrapperConfig[], resultArr: string[] = []): string[] => {
  let prevElement: StageElementWrapperConfig
  stages?.forEach(stage => {
    if (!prevElement) {
      prevElement = stage
      return
    }
    if (stage.parallel) {
      getParallelNodeLinks(stage.parallel, prevElement, resultArr)
    } else {
      resultArr.push(
        getFinalSVGArrowPath('tree-container', prevElement.stage?.identifier as string, stage.stage?.identifier, false)
      )
    }
    prevElement = stage
  })
  return resultArr
}
const getParallelNodeLinks = (
  stages: StageElementWrapperConfig[],
  prevStage: StageElementWrapperConfig,
  resultArr: string[] = []
): void => {
  const [firstStage, ...restStages] = stages
  resultArr.push(
    getFinalSVGArrowPath('tree-container', prevStage.stage?.identifier as string, firstStage.stage?.identifier, false)
  )
  restStages?.forEach(stage => {
    resultArr.push(
      getFinalSVGArrowPath('tree-container', firstStage.stage?.identifier as string, stage.stage?.identifier, true)
    )
  })
}

const getScaleToFitValue = (elm: HTMLElement, paddingFromBottom = 20): number => {
  return (
    1 /
    Math.max(
      elm.clientWidth / window.innerWidth,
      elm.clientHeight / (window.innerHeight - elm.offsetTop - paddingFromBottom)
    )
  )
}
export {
  getFinalSVGArrowPath,
  setupDragEventListeners,
  getSVGLinksFromPipeline,
  ZOOM_INC_DEC_LEVEL,
  INITIAL_ZOOM_LEVEL,
  getScaleToFitValue
}
