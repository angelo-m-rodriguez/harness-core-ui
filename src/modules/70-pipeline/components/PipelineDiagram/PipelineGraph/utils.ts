import { defaultTo } from 'lodash-es'
import type { PipelineGraphState } from '../types'

const shouldAttachRef = (
  intersectingIndex: number,
  isCurrentChildLast: boolean,
  indexRelativeToParent: number
): boolean => {
  return intersectingIndex === -1 // intersecting index is not set
    ? isCurrentChildLast // current child is not the last child in parallel
    : intersectingIndex === indexRelativeToParent // intersecting index is not equal to indexRelativeToParent (i.e index+1 as parent is 0)
}

const shouldRenderGroupNode = (attachRef: boolean, isCurrentChildLast: boolean): boolean => {
  return attachRef && !isCurrentChildLast // render group node is ref is attached and current node is not the last child
}

const isFirstNodeAGroupNode = (intersectingIndex: number, collapseOnIntersect = false, childLength = 0): boolean =>
  intersectingIndex === 0 && collapseOnIntersect && childLength > 0 // if intersecting index is 0 and collapseOnIntersect is enabled

const showChildNode = (indexRelativeToParent: number, intersectingIndex: number): boolean => {
  return !(indexRelativeToParent > intersectingIndex && intersectingIndex !== -1) // child index is greater than intersectingIndex and intersectingIndex is set to some value
}

const isNodeParallel = (node: PipelineGraphState): boolean => defaultTo(node?.children?.length, 0) > 0

export { shouldAttachRef, shouldRenderGroupNode, isFirstNodeAGroupNode, showChildNode, isNodeParallel }
