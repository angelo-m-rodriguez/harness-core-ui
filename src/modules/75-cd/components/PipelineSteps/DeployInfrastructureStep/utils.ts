/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'

import { getMultiTypeFromValue, MultiTypeInputType, SelectOption } from '@harness/uicore'
import type { EnvironmentResponseDTO, PipelineInfrastructure } from 'services/cd-ng'

import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'

export interface PipelineInfrastructureV2 extends PipelineInfrastructure {
  environmentOrEnvGroupRef?: SelectOption
  environmentGroup?: any
  environmentRef2?: any
  infrastructureRef?: any
}

export interface DeployInfrastructureProps {
  initialValues: PipelineInfrastructureV2
  onUpdate?: (data: PipelineInfrastructureV2) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: PipelineInfrastructureV2
    path?: string
    readonly?: boolean
  }
}

export interface DeployInfrastructureState {
  isEdit: boolean
  isEnvironment: boolean
  formik?: FormikProps<PipelineInfrastructureV2>
  data?: EnvironmentResponseDTO
}

export function isEditEnvironment(data: PipelineInfrastructureV2): boolean {
  if (
    getMultiTypeFromValue(data.environmentRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.environmentOrEnvGroupRef)
  ) {
    return true
  }
  return false
}

export function isEditInfrastructure(data: any): boolean {
  if (
    getMultiTypeFromValue(data.infrastructureRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.infrastructureRef)
  ) {
    return true
  } else if (data.infrastructure && !isEmpty(data.infrastructure.identifier)) {
    return true
  }
  return false
}
