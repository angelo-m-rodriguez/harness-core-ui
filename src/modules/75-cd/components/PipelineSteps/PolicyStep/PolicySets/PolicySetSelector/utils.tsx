/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { FontVariation, Layout, Text } from '@harness/uicore'
import { useStrings } from 'framework/strings'

export const PolicySetSelectorDialogTitle = (): JSX.Element => {
  const { getString } = useStrings()
  return (
    <Layout.Horizontal
      spacing="xsmall"
      padding={{ top: 'xxlarge', left: 'medium', right: 'large', bottom: 'medium' }}
      flex={{ justifyContent: 'space-between' }}
    >
      <Layout.Vertical>
        <Text font={{ variation: FontVariation.H3 }}>{getString('common.policiesSets.selectPolicySet')}</Text>
      </Layout.Vertical>
    </Layout.Horizontal>
  )
}
