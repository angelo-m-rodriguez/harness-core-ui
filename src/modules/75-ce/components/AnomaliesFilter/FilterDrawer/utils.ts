/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiSelectOption } from '@harness/uicore'
import { omit } from 'lodash-es'
import type { AnomalyFilterProperties } from 'services/ce'
import type { AnomaliesFilterFormType } from './FilterDrawer'

const getMultiSelectOptions = (values: string[]): MultiSelectOption[] =>
  values.map(item => ({ label: item, value: item }))

const getValueFromOption = (values: MultiSelectOption[]): string[] => values.map(item => item.value.toString())

export const getAnomalyFormValuesFromFilterProperties = (
  filterProperties: AnomalyFilterProperties
): AnomaliesFilterFormType => {
  const formValues: AnomaliesFilterFormType = {}

  Object.entries(omit(filterProperties, 'filterType')).forEach(([key, value]) => {
    if (value) {
      if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
        formValues[key] = +value
      } else {
        ;(formValues as any)[key] = getMultiSelectOptions(value as string[])
      }
    }
  })

  return formValues
}

export const getAnomalyFilterPropertiesFromForm = (formData: AnomaliesFilterFormType): AnomalyFilterProperties => {
  const filterProperties: AnomalyFilterProperties = { filterType: 'Anomaly' }

  Object.entries(formData).forEach(([key, value]) => {
    if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
      filterProperties[key] = +value
    } else {
      ;(filterProperties as any)[key] = getValueFromOption(value)
    }
  })

  return filterProperties
}
