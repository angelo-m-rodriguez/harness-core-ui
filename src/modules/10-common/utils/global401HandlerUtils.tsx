/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { History } from 'history'
import routes from '@common/RouteDefinitions'
import SecureStorage from 'framework/utils/SecureStorage'
import { getLoginPageURL } from 'framework/utils/SessionUtils'
import { returnUrlParams } from './routeUtils'

export const global401HandlerUtils = (history: History) => {
  SecureStorage.clear()
  history.push({
    pathname: routes.toRedirect(),
    search: returnUrlParams(getLoginPageURL({ returnUrl: window.location.href }))
  })
}
