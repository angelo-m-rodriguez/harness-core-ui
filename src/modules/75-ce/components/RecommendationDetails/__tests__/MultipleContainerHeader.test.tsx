/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'

import MultipleContainerHeader from '../MultipleContainerHeader'

const setContainerVisibleMock = jest.fn()

describe('test cases for multiple conatiner header', () => {
  test('Multiple containers/Container visible', async () => {
    const { container } = render(
      <TestWrapper>
        <MultipleContainerHeader
          containerName="Test Container"
          containerVisible={true}
          currentContainer={1}
          totalContainers={10}
          toggleContainerVisble={setContainerVisibleMock}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('Single container/Container always visible', async () => {
    const { container } = render(
      <TestWrapper>
        <MultipleContainerHeader
          containerName="Test Container"
          containerVisible={true}
          currentContainer={1}
          totalContainers={1}
          toggleContainerVisble={setContainerVisibleMock}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('Multiple containers/Container not visible', async () => {
    const { container } = render(
      <TestWrapper>
        <MultipleContainerHeader
          containerName="Test Container"
          containerVisible={false}
          currentContainer={1}
          totalContainers={10}
          toggleContainerVisble={setContainerVisibleMock}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('Single container/Container not visible', async () => {
    const { container } = render(
      <TestWrapper>
        <MultipleContainerHeader
          containerName="Test Container"
          containerVisible={false}
          currentContainer={1}
          totalContainers={1}
          toggleContainerVisble={setContainerVisibleMock}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()

    expect(container.querySelector('[class*="toggleContainerIcon"]')).toBeNull()
  })
})
