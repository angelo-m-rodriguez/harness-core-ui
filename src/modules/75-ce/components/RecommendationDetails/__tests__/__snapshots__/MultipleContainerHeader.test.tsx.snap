// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for multiple conatiner header Multiple containers/Container not visible 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main containerNameCard StyledProps--padding-medium StyledProps--background-primary1"
  >
    <div
      class="StyledProps--font StyledProps--main"
    >
      <span
        class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey800 StyledProps--font-variation-body2"
      >
        ce.recommendation.detailsPage.containerHeader
      </span>
      <span
        class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey800 StyledProps--font-variation-body"
      >
        Test Container
      </span>
    </div>
    <span
      class="bp3-icon bp3-icon-caret-down StyledProps--main toggleContainerIcon StyledProps--color StyledProps--color-primary5"
      icon="caret-down"
    >
      <svg
        data-icon="caret-down"
        height="18"
        viewBox="0 0 16 16"
        width="18"
      >
        <desc>
          caret-down
        </desc>
        <path
          d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`test cases for multiple conatiner header Multiple containers/Container visible 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main containerNameCard StyledProps--padding-medium StyledProps--background-primary1"
  >
    <div
      class="StyledProps--font StyledProps--main"
    >
      <span
        class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey800 StyledProps--font-variation-body2"
      >
        ce.recommendation.detailsPage.containerHeader
      </span>
      <span
        class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey800 StyledProps--font-variation-body"
      >
        Test Container
      </span>
    </div>
    <span
      class="bp3-icon bp3-icon-caret-up StyledProps--main toggleContainerIcon StyledProps--color StyledProps--color-primary5"
      icon="caret-up"
    >
      <svg
        data-icon="caret-up"
        height="18"
        viewBox="0 0 16 16"
        width="18"
      >
        <desc>
          caret-up
        </desc>
        <path
          d="M11.87 9.17s.01 0 0 0l-3.5-4C8.28 5.07 8.15 5 8 5s-.28.07-.37.17l-3.5 4a.495.495 0 00.37.83h7a.495.495 0 00.37-.83z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`test cases for multiple conatiner header Single container/Container always visible 1`] = `<div />`;

exports[`test cases for multiple conatiner header Single container/Container not visible 1`] = `<div />`;
